packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -100,117 +101,80 @@ export const areRenderContextsEqual = (
   );
 };
 
-interface UseGridVirtualScrollerProps {
-  ref: React.Ref<HTMLDivElement>;
-  renderZoneMinColumnIndex?: number;
-  renderZoneMaxColumnIndex?: number;
-  onRenderZonePositioning?: (params: { top: number; left: number }) => void;
-  getRowProps?: (id: GridRowId, model: GridRowModel) => any;
-}
+const EMPTY_RENDER_CONTEXT = {
+  firstRowIndex: 0,
+  lastRowIndex: 0,
+  firstColumnIndex: 0,
+  lastColumnIndex: 0,
+};
 
-interface ContainerDimensions {
-  width: number | null;
-  height: number | null;
-}
+export const EMPTY_PINNED_COLUMNS = {
+  left: [] as GridStateColDef[],
+  right: [] as GridStateColDef[],
+};
+
+export const EMPTY_DETAIL_PANELS = Object.freeze(new Map<GridRowId, React.ReactNode>());
 
-// The `maxSize` is 3 so that reselect caches the `renderedColumns` values for the pinned left,
-// unpinned, and pinned right sections.
-const MEMOIZE_OPTIONS = { maxSize: 3 };
+export type VirtualScroller = ReturnType<typeof useGridVirtualScroller>;
 
-export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
+export const useGridVirtualScroller = () => {
```

Comment:
I've noticed column headers jump and misalignment that seem to be caused by column virtualization:
https://deploy-preview-10059--material-ui-x.netlify.app/x/react-data-grid/performance/#visualization


https://github.com/mui/mui-x/assets/13808724/c706def2-1e8c-4690-aa67-1ab31b197d3b

Reply:
In this demo, the column headers offset persists (the number in the column header should match the second number in the cell): https://deploy-preview-10059--material-ui-x.netlify.app/x/react-data-grid/virtualization/#column-virtualization

<img width="846" alt="Screenshot 2023-12-05 at 17 39 44" src="https://github.com/mui/mui-x/assets/13808724/b1c5d79c-48c7-4e55-94d4-646613b82ed4">

Reply:
The header border is also not stretching and goes away when scrolling horizontally.

Reply:
Yes, I tried to show this on the video, but didn't mention it as it seemed like a minor issue in this context 😀